### PR TITLE
Apply rkt-io changes to vanilla LKL

### PR DESCRIPTION
**Pull request by wamos**
Though I can't directly spin up rkt-io repo. Their LKL modification can be reused for our LKL+DPDK development. The changes here span intend to work for both SPDK and DPDK. I don't see the harm of including SPDK related changes as a version of LKL.

### CODE REVIEW

File: arch/lkl/mm/bootmem.c
Comment (wamos) [inline arch/lkl/mm/bootmem.c:1]: This is SGX related code

File: arch/lkl/mm/bootmem.c [hunk 1]
```diff
@@ -1,19 +1,79 @@
...
+		spdk_gfp_flags = 0;
+		spdk_slab_flags = 0;
+	} else {
+		init_dma_mem(DMA_ZONE_SPDK, spdk_dma_memory_begin,
+			     spdk_dma_memory_end);
+		init_dma_mem(DMA_ZONE_DPDK, dpdk_dma_memory_start,
+			     dpdk_dma_memory_end);
+	}
+
+	mem_size = 1024 * 1024 * 256;
```
Comment (wamos) [inline arch/lkl/mm/bootmem.c:72]: `mem_size = 1024*1024*256` is SGX related

File: arch/lkl/mm/bootmem.c [hunk 2]
```diff
@@ -1,19 +1,79 @@
...
+		zones_size[ZONE_NORMAL] = (mem_size) >> PAGE_SHIFT;
+		free_area_init_node(zone, zones_size, min_low_pfn, NULL);
+	}
+}
+
 void __init bootmem_init(unsigned long mem_sz)
 {
 	unsigned long bootmap_size;
 
-	mem_size = mem_sz;
```
Comment (plaublin) [inline arch/lkl/mm/bootmem.c:14]: We want to keep this line.

File: arch/lkl/mm/bootmem.c [hunk 3]
```diff
@@ -1,19 +1,79 @@
...
+	if (strstr(boot_command_line, "spdk_dma_alloc=no")) {
+		spdk_gfp_flags = 0;
```
Comment (plaublin) [inline arch/lkl/mm/bootmem.c:63]: We don't need this.

**Pull request merged**